synths/core-synths.scd

```diff
@@ -52,6 +50,47 @@ live coding them requires that you have your SuperDirt instance in an environmen
 		}, [\ir, \ir, \ir, \ir, \ir, \ir, \ir, \ir]).add;
 	};
 
+	// this is the time-stretching sample player
+	(1..SuperDirt.maxSampleNumChannels).do { |sampleNumChannels|
+
+		var name = format("dirt_stretchsample_%_%", sampleNumChannels, numChannels);
+
+		SynthDef(name, { |out, bufnum, sustain = 1, begin = 0, end = 1, speed = 1, endSpeed = 1, freq = 440, pan = 0, timescale = 1|
+
+			var sound, rate, phase, sawrate, numFrames, index, windowIndex, window, timescaleStep, windowSize;
+
+			// playback speed
+			rate = Line.kr(speed, endSpeed, sustain) * (freq / 60.midicps);
+
+			// sample phase
+			// BufSampleRate adjusts the rate if the sound file doesn't have the same rate as the soundcard
+			//phase =  Sweep.ar(1, rate * BufSampleRate.ir(bufnum)) + (BufFrames.ir(bufnum) * begin);
+
+			numFrames = BufFrames.ir(bufnum);
+			windowSize = numFrames * timescale / 29.0;
+			windowSize = windowSize.clip(1000.0*timescale, 4410.0);
+			timescaleStep = windowSize / timescale;
+			sawrate = rate * BufSampleRate.ir(bufnum) / (absdif(begin, end) * timescale * numFrames);
+			phase = (speed.sign * LFSaw.ar(sawrate, 1)).range(begin,end) * numFrames * timescale;
+			index = (phase.div(timescaleStep) - [1.0, 0.0]) * (timescaleStep / timescale - timescaleStep) + phase;
+			windowIndex = phase - (phase.div(timescaleStep) - [1.0, 0.0] * timescaleStep);
+			window = exp(-0.5 * (windowIndex/windowSize - 0.5/0.1)**2);
```

Comment:
this can be written as:
`exp((5 * windowIndex/windowSize - 2.5) ** 2)`

or maybe better as:
`exp(squared((5 * windowIndex/windowSize - 2.5)))`

why these numbers and not others?

Reply:
It's a gaussian centered at the middle of the window with a width (sigma) of 0.1 the window size.  The 0.1, though, is really just trial and error to find what sounds good for a wide variety of sounds.  But can definitely combine this into fewer operations.

Reply:
maybe it would make sense as a parameter that you can pass, with a good default (in any case it makes sense to avoid unnecessary operations on a signal).